### PR TITLE
Check for known block when processing "header first"

### PR DESCRIPTION
**Pull request by antiochp**
When processing a header in "header first" propagation we currently perform a quick check to see if we have just processed this header.
This does not cover the case where a peer sends (intentionally or otherwise) an old header.

We should do the same check here that we already do when processing _full_ blocks - 
* Is the block for this header on the head of the current chain?
* Is the block for this header in our set of orphan blocks?
* Is the block for this header in our database?

If this is a header for a "known" block then we should halt processing of the header.

### CODE REVIEW

File: chain/src/pipe.rs [hunk 1]
```diff
@@ -260,19 +260,10 @@ pub fn process_block_header(header: &BlockHeader, ctx: &mut BlockContext<'_>) ->
...
-	Ok(())
-}
+	// Check if this header is already "known" from processing a previous block.
+	check_known(header, ctx)?;
 
-/// Quick in-memory check to fast-reject any block header we've already handled
-/// recently. Keeps duplicates from the network in check.
-/// ctx here is specific to the header_head (tip of the header chain)
-fn check_header_known(header: &BlockHeader, ctx: &mut BlockContext<'_>) -> Result<(), Error> {
-	let header_head = ctx.batch.header_head()?;
```
Comment (garyyu) [inline chain/src/pipe.rs]: It said `ctx here is specific to the header_head (tip of the header chain)`.

but in `check_know()`, the `check_known_head(header, ctx)` is using `ctx.batch.head()`, not `header_head`.

Comment (antiochp) [reply chain/src/pipe.rs]: This code is deleted.

Comment (garyyu) [reply chain/src/pipe.rs]: I mean your new code has very different behavior from the old `check_header_known` which use header_head.  now you're using `ctx.batch.head()`.

Comment (antiochp) [reply chain/src/pipe.rs]: We used to check for the header itself was known based on the header chain.
The only time we would know about headers and not yet now about the full blocks is during initial sync.
So it's safe to check if we know about the header based on the full chain - we're processing "header first" in order to process full blocks.

**Pull request merged**